src/TypeScript/Rules/RuleDefinitionsProvider.cs

```diff
@@ -20,33 +20,22 @@
 
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
 using System.Text;
 using Newtonsoft.Json;
 
 namespace SonarLint.VisualStudio.TypeScript.Rules
 {
-    internal interface ITypeScriptRuleDefinitionsProvider
-    {
-        /// <summary>
-        /// Returns the metadata descriptions for all TypeScript rules
-        /// </summary>
-        IEnumerable<RuleDefinition> GetDefinitions();
-    }
-
-    internal interface IJavaScriptRuleDefinitionsProvider
-    {
-        /// <summary>
-        /// Returns the metadata descriptions for all JavaScript rules
-        /// </summary>
-        IEnumerable<RuleDefinition> GetDefinitions();
-    }
-
     [Export(typeof(ITypeScriptRuleDefinitionsProvider))]
     [Export(typeof(IJavaScriptRuleDefinitionsProvider))]
+    [Export(typeof(ITypeScriptRuleKeyMapper))]
+    [Export(typeof(IJavaScriptRuleKeyMapper))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    internal class RuleDefinitionsProvider : ITypeScriptRuleDefinitionsProvider, IJavaScriptRuleDefinitionsProvider
+    internal class RuleDefinitionsProvider :
```

Comment:
It's a bit confusing that we have 2 file names that are nearly identical. Maybe we should rename this one, i.e. something like `EslintBridgeRulesRepository`, or `JsTsRulesContainer`, etc.